src/cljs/athens/events.cljs

```diff
@@ -497,23 +499,26 @@
     (indent uid)))
 
 
-;; TODO: no-op when user tries to unindent to a child out of current context
 (defn unindent
-  [uid]
+  [uid context-root]
   (let [parent (db/get-parent [:block/uid uid])
         grandpa (db/get-parent (:db/id parent))
         new-block {:block/uid uid :block/order (inc (:block/order parent))}
         reindex-grandpa (->> (inc-after (:db/id grandpa) (:block/order parent))
                              (concat [new-block]))]
-    (when (and parent grandpa)
-      {:transact! [[:db/retract (:db/id parent) :block/children [:block/uid uid]]
-                   {:db/id (:db/id grandpa) :block/children reindex-grandpa}]})))
+    (if (= (:block/uid parent) context-root) ; if the parent node is the context-root, prevent unindent
+      {}
+      (when (and parent grandpa)
+        {:transact! [[:db/retract (:db/id parent) :block/children [:block/uid uid]]
+                     {:db/id (:db/id grandpa) :block/children reindex-grandpa}]}))))
```

Comment:
Flatten/refactor this nested conditional expression.

I'm not sure we even need `when (and parent grandpa)` anymore since `context-root` is a stricter check.

Reply:
Yeah it doesn't seem like we do. `context-root-uid` seems to handle the case just fine.

Reply:
> I'm not sure we even need when (and parent grandpa) anymore since context-root is a stricter check.

Good point Jeff 🧠